howso/client/protocols.py

```diff
@@ -1,18 +1,162 @@
-import typing as t
+from typing import (
+    Any,
+    Dict,
+    List,
+    Protocol,
+    Optional,
+    runtime_checkable,
+    TYPE_CHECKING
+)
 
-if t.TYPE_CHECKING:
+from howso.direct.core import HowsoCore
+from howso.client.cache import TraineeCache
+
+if TYPE_CHECKING:
+    from howso.openapi.models import (
+        Metrics,
+        Trainee,
+        TraineeInformation,
+    )
     from howso.openapi.models import Project
 
 __all__ = [
     "ProjectClient",
 ]
 
 
-@t.runtime_checkable
-class ProjectClient(t.Protocol):
+@runtime_checkable
+class BaseClientProtocol(Protocol):
```

Comment:
Seems like the actual HowsoDirectClient itself could serve our needs without defining this Protocol.